### ISSUE TITLE
Initialization of the Apache HttpAsyncClient

### ISSUE THREAD

**Issue by kriszman** [a-schild/nextcloud-java-api#55]
Hi,

I've been successfully using this library for some weeks now, it really helped me a lot, however, recently I stumbled across a feature that I'm missing somehow. It's about missing configuration options related to the initialization of the http client that you're using. The only customization that you currently provide is about SSL, so I've been wondering if that's been chosen intentionally. If not I'd like to change the current default initialization of the http client in order to be able to provide system properties to the http client. The main reason behind this is the fact that I'd like to be able to tune the number of concurrent requests, which is possible using system properties, which, though, have to be activated during initialization of the client.
Long story short, the question is if it's worth doing so and whether or not you would eventually accept a corresponding pull request.

Thx a lot!
Best regards,
Christian

**Comment by a-schild** [comment]
Hello Christian,

yes, I would accept a pull request for this.
But please add a test case for this and some examples.

André

**Comment by kriszman** [comment]
Hi André,

great, thx for your feedback!

BR,
Christian
